### PR TITLE
Add Effect.makeAbortSignal

### PR DESCRIPTION
**Pull request by jessekelly881**
Exports existing makeAbortSignal

### CODE REVIEW

File: packages/effect/src/Effect.ts [hunk 1]
```diff
@@ -5392,3 +5392,27 @@ export const Tag: <const Id extends string>(id: Id) => <Self, Type>() =>
...
   }
+
+/**
+ * Creates an `AbortSignal`.
+ * @since 3.1.0
+ */
+export const makeAbortSignal: Effect<AbortSignal> = map(
+  tap(
+    sync(() => new AbortController()),
+    (controller) => onInterrupt(() => sync(() => controller.abort()))
```
Comment (tim-smart) [inline packages/effect/src/Effect.ts:5403]: This doesn't make sense, as the onInterrupt effect is only applied regionally / to the effect it wraps.

File: packages/effect/src/Effect.ts [hunk 2]
```diff
@@ -5392,3 +5392,27 @@ export const Tag: <const Id extends string>(id: Id) => <Self, Type>() =>
...
+  ),
+  (_) => _.signal
+)
+
+/**
+ * Creates an `AbortSignal` that is automatically aborted when the Scope is closed.
+ * @since 3.1.0
+ */
+export const makeAbortSignalScoped: Effect<AbortSignal, never, Scope.Scope> = map(
```
Comment (tim-smart) [inline packages/effect/src/Effect.ts:5412]: Hmm will think about it. There aren't many cases where this is useful - usually the signals passed to tryPromise / async are all you need.

File: packages/effect/src/Effect.ts (hunk 1 above)
Comment (jessekelly881) [reply packages/effect/src/Effect.ts:5403]: Oh interesting. Ok. I'll remove this.

File: packages/effect/src/Effect.ts (hunk 2 above)
Comment (jessekelly881) [reply packages/effect/src/Effect.ts:5412]: Typically async and  promise are enough. But there have been a few circumstances where I've needed a signal inside of an Effect.gen for example and using async/promise didn't make sense.

**Comment by tim-smart** [comment]
Considering this would no longer be used internally, I don't think we need this addition.

If you feel strongly about adding it, feel free to re-open to discuss.

**Pull request closed without merging**